libs/design-system-core/src/components/Button/button-icon.tsx

```diff
@@ -38,7 +38,7 @@ export const ButtonIcon: React.FC<
     <Icon
       icon={icon}
       solid={solid}
-      size={size}
+      size={size === 'xs' ? 'sm' : size}
```

Comment:
hmm, i wonder why is acrobatics required. what if we pass the iconSize as a separate prop?

Reply:
this is for the icon only button when the passed size if very small 'xs'. rather than specifying a new set of width/height for that singular use case.